### ISSUE TITLE
Cast string to multiple values for a field

### ISSUE THREAD

**Issue by omaralweli** [metabase/metabase#37215]
**Is your feature request related to a problem? Please describe.**
We are trying to display set of data in a field but it seems that Metabase is not recognising the set which is a number of strings separated by a semicolon. 

**Describe the solution you'd like**
It would be great if there is a native support for the set described above but i'll settle for an option to cast this specific data type.

**Describe alternatives you've considered**
I considered doing some magic behind the scene to manipulate the data before mapping it to Metabase but I am honestly not sure if a different type of set can be read by Metabase.

**How important is this feature to you?**
It is fairly important because we are forced to avoid using a datatype with Metabase that we otherwise fully support.

**Additional context**
The issue was discovered because we have a custom mapping for the values and have trouble mapping that to the string.

Basically we are trying to map this data :
![Screenshot 2024-01-03 at 11 41 17](https://github.com/metabase/metabase/assets/77667105/8524c8ae-90da-4e6a-8087-b7ea17cca2a8)

But this is what Metabase display:
![Screenshot 2024-01-03 at 09 28 41](https://github.com/metabase/metabase/assets/77667105/3ce2ac01-ec6f-48d8-9694-61374cfaa7d4)

**Comment by paoliniluis** [comment]
Is https://github.com/metabase/metabase/issues/18974 what you're looking for?

**Comment by omaralweli** [comment]
> Is #18974 what you're looking for?

I believe that feature doesn't yet cover the need for mapping multiple options but the idea is the same.
